### PR TITLE
ghc: Update to version 8.8.3

### PR DESCRIPTION
**Pull request by essandess**
ghc: Update to version 8.8.3

Fixes: https://trac.macports.org/ticket/59467

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/ghc/Portfile [hunk 1]
```diff
@@ -123,11 +125,15 @@ if { [variant_isset "prebuilt"] } {
 
     # add the bootstrap binary and the sphinx-build binary to PATH
     configure.env-append \
-        PATH=$env(PATH):${bootstrap_dir}/bin:${frameworks_dir}/Python.framework/Versions/${python3_version}/bin
+        GHC=${bootstrap_dir}/bin/ghc \
+        "PATH=/usr/bin:$env(PATH):${bootstrap_dir}/bin:${frameworks_dir}/Python.framework/Versions/${python3_version}/bin"
+
+    # Fix Xcode 11 issue; see https://trac.macports.org/ticket/59467
+    compiler.blacklist-append clang
```
Comment (kencu) [inline lang/ghc/Portfile]: Steve, are you sure that you need to blacklist clang? The issue in ticket 59467 was all about a definition that was not to be used on Catalina <https://gitlab.haskell.org/ghc/ghc/merge_requests/2190/diffs?commit_id=c6fca740a708ad9db5fdb595ad06f727289cdc07> and had nothing to do with Xcode 11...

Comment (essandess) [reply lang/ghc/Portfile]: Thanks! Removed.

**Comment by essandess** [comment]
Note: I’m unable to get `port test ghc` working. I keep hitting this age-old `_iconv` linking issue. IIRC, I’ve had this working in the past. I’m not sure if the problem is general or my local setup.

If anyone knows how to address this, please chime in.

I’d say to merge the new version and address test issues later.

Here’s the error I see:

```
:info:test clang: error: linker command failed with exit code 1 (use -v to see invocation)
:info:test `gcc' failed in phase `Linker'. (Exit code: 1)
:info:test /Applications/Xcode.app/Contents/Developer/usr/bin/make -C ./tests test
:info:test "/opt/local/var/macports/build/_opt_local_ports_lang_ghc/ghc/work/src/ghc-8.8.3/inplace/bin/ghc-stage2" --make -o ../mk/ghc-config ../mk/ghc-config.hs
:info:test [1 of 1] Compiling Main             ( ../mk/ghc-config.hs, ../mk/ghc-config.o ) [flags changed]
:info:test Linking ../mk/ghc-config ...
:info:test Undefined symbols for architecture x86_64:
:info:test   "_iconv", referenced from:
:info:test       _hs_iconv in libHSbase-4.13.0.0.a(iconv.o)
```

**Comment by kencu** [comment]
pita issue. can you just deactivate macports iconv for purposes of running the tests?

**Comment by essandess** [comment]
> pita issue. can you just deactivate macports iconv for purposes of running the tests?

Yeah, `port test ghc` works if port `libiconv` Is deactivated. Thanks. (But no other MacPorts command works while it’s deactivate, including `git` 😝) 

I updated the Portfile with working test directives, so long as `libiconv` is deactivated.

**Comment by kencu** [comment]
Is this ready for committing?

**Comment by mojca** [comment]
Whitespace changes should go to a separate commit.
Please test asymptote build.

**Comment by essandess** [comment]
@kencu Yes, this is ready to commit.

@mojca I reverted the white space changes. I don’t understand what `asymptote` has to do with this PR.

**Comment by mojca** [comment]
Asymptote doesn't have anything to do with this PR per se, but `ghc` is one of its dependencies and it's sometimes relatively picky about the version of `ghc`, particularly on macOS, FreeBSD, cygwin.

**Comment by essandess** [comment]
> Asymptote doesn't have anything to do with this PR per se, but `ghc` is one of its dependencies and it's sometimes relatively picky about the version of `ghc`, particularly on macOS, FreeBSD, cygwin. So it would be helpful to just make sure that it still compiles.

Port `asymptote` doesn’t depend at all on `ghc`. This yields a goose egg:
```
port rdeps asymptote | grep ghc
```

Even if it did, this is a required update to fix port `ghc`. If it breaks something else, then that something else will have to be fixed downstream.

But I don’t see how this relates to `asymptote` at all.

**Comment by mojca** [comment]
I'm sorry, that was stupid of me. That is `gc` rather that `ghc`. I'm extremely sorry.

**Comment by essandess** [comment]
> I'm sorry, that was stupid of me. That is `gc` rather that `ghc`. I'm extremely sorry.

Not at all, thanks for the input.

**Pull request merged**